### PR TITLE
cron: adding validation for range or hyphenated inputs for start and end

### LINKED ISSUE #1809: Cron: no validation for range input in start an end field in metadata
### Report

Cron standard include range as an input too. When i tried : 

> End:              18 * * * *
> Start:             11-13 * * * *

SO created with no error. Scaling happened with start at 13 and end at 18 minutes.
or 

> when start: -32 * * * * 

It started containers with terminating state and deployment available number become 0.  No error in operator or scaledObject log.
There should be validation for range value input too. May I ask , are we taking different  start and end inputs rather than range intentionally? How about a different field for eg. Range? 

### Expected Behavior

There should be validation and set error for no support for range in SO events or there should be provision for using range inputs

### Actual Behavior

Cron syntax with range as an input too. When i tried : 

> End:              18 * * * *
> Start:             11-13 * * * *

SO created with no error. Scaling happened with start at 13 and end at 18 minutes.
or 

> when start: -32 * * * * 

It started containers with terminating state and deployment available number become 0. No error in scaledobject events or operator logs

### Steps to Reproduce the Problem

1. use start  Start:    11-13 * * * * & end : 18 * * * * for first case & for second case- use start: -13 * * * * in cron scaledObject trigger metadata
2. apply SO and check the logs in `kubectl describe so`
3. Check output log from `kubectl logs -n keda keda-operator-xxxx-xxxx`

### Logs from KEDA operator

No error occurred/mentioned though desired replicacount not achieved. the scaling is acc. to min replicacount

### KEDA Version

2.2.0

### Kubernetes Version

1.20

### Platform

Other

### Scaler Details

Cron

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Ritikaa96**
Signed-off-by: Ritikaa96 <ritika@india.nec.com>

Adding validation to check start and end for range values such as : 

> start: "30-39 * * * *" 

or negative values such as 

> start: "-39 * * * *"

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added

Fixes #1809

### CODE REVIEW

**Review by zroubalik — COMMENTED** [review]
Looking good, thanks for this fix! Could you please add a few tests for metadata parsing?

**Comment by Ritikaa96** [comment]
> Looking good, thanks for this fix! Could you please add a few tests for metadata parsing?

Sure @zroubalik

**Comment by Ritikaa96** [comment]
Hi, @zroubalik . Added the tests for metadata parsing. PTAL

File: pkg/scalers/cron_scaler_test.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ var testCronMetadata = []parseCronMetadataTestData{
 	{validCronMetadata, false},
 	{map[string]string{"timezone": "Asia/Kolkata", "start": "30 * * * *", "end": "45 * * * *"}, true},
 	{map[string]string{"start": "30 * * * *", "end": "45 * * * *", "desiredReplicas": "10"}, true},
+	{map[string]string{"timezone": "Asia/Kolkata","start": "30-33 * * * *", "end": "45 * * * *", "desiredReplicas": "10"}, true},
+	{map[string]string{"timezone": "Asia/Kolkata","start": "30 * * * *", "end": "45-50 * * * *", "desiredReplicas": "10"}, true},
```
Comment (zroubalik) [inline pkg/scalers/cron_scaler_test.go:35]: Could you please add some incorrect entries, so we can be sure that the validation is working correctly? Thanks!

Comment (Ritikaa96) [reply pkg/scalers/cron_scaler_test.go:35]: Hi @zroubalik , the added entries are incorrect as they have range values. Could you please elaborate what other incorrect entries do you expect? Are these good: 

> {map[string]string{"timezone": "Asia/Kolkata","start": "-33 * * * *", "end": "45 * * * *", "desiredReplicas": "10"}

> {map[string]string{"timezone": "Asia/Kolkata","start": "30 * * * *", "end": " - * * * *", "desiredReplicas": "10"}

> {map[string]string{"timezone": "Asia/Kolkata","start": "33 * * * *", "end": "45 * * -3 *", "desiredReplicas": "10"}

Comment (zroubalik) [reply pkg/scalers/cron_scaler_test.go:35]: Yeah, these 3 are good!

**Comment by Ritikaa96** [comment]
@zroubalik PTAL

**Comment by zroubalik** [comment]
> @zroubalik PTAL

Seems like the linter complains about the code. Could you please check?

**Review by zroubalik — APPROVED** [review]
LGTM, thanks @Ritikaa96 !

**Pull request merged**